tests/vanilla/utils/atomWithReset.test.ts

```diff
@@ -0,0 +1,48 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { RESET, atomWithReset } from 'jotai/vanilla/utils'
+
+vi.mock('jotai/vanilla', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('jotai/vanilla')>()
+  return {
+    ...actual,
+    atom: vi.fn(actual.atom),
+  }
+})
+
+describe('atomWithReset', () => {
+  let initialValue: number
+  let testAtom: any
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    initialValue = 10
+    testAtom = atomWithReset(initialValue)
+  })
+
+  it('should create an atom with initial value', () => {
+    const { init } = testAtom
+    expect(init).toBe(initialValue)
+  })
+
+  it('should reset to initial value using RESET', () => {
+    const set = vi.fn()
+    const get = vi.fn(() => 20)
+    testAtom.write(get, set, RESET)
```

Comment:
`write` is an implementation detail, and we would like to test with store api.
can you try this?
```js
const store = createStore();
store.set(testAtom, RESET);
```

Reply:
@dai-shi I have already added the test code for using `createStore` and kept the previous test code, as it can isolate the dependency on `createStore`.

Reply:
It's testing the internal behavior, which isn't guaranteed to work as it is now (it's likely though), so let's remove it.

Reply:
OK